openbas-api/src/main/java/io/openbas/xtmhub/XtmHubService.java

```diff
@@ -24,11 +30,102 @@ public PlatformSettings register(@NotBlank final String token) {
         LocalDateTime.now(),
         XtmHubRegistrationStatus.REGISTERED,
         currentUser.getId(),
-        currentUser.getName());
+        currentUser.getName(),
+        LocalDateTime.now(),
+        true);
   }
 
   public PlatformSettings unregister() {
     return this.platformSettingsService.updateXTMHubRegistration(
-        null, null, XtmHubRegistrationStatus.UNREGISTERED, null, null);
+        null, null, XtmHubRegistrationStatus.UNREGISTERED, null, null, null, null);
+  }
+
+  public PlatformSettings refreshConnectivity() {
+    PlatformSettings settings = platformSettingsService.findSettings();
+
+    if (!isRegisteredWithXtmHub(settings)) {
+      return settings;
+    }
+
+    ConnectivityCheckResult checkResult = checkConnectivityStatus(settings);
+    handleConnectivityLossNotification(settings, checkResult);
+
+    return updateRegistrationStatus(settings, checkResult);
+  }
+
+  private boolean isRegisteredWithXtmHub(PlatformSettings settings) {
+    return StringUtils.isNotBlank(settings.getXtmHubToken());
+  }
+
+  private ConnectivityCheckResult checkConnectivityStatus(PlatformSettings settings) {
+    XtmHubConnectivityStatus status =
+        xtmHubClient.refreshRegistrationStatus(
+            settings.getPlatformId(), settings.getPlatformVersion(), settings.getXtmHubToken());
+
+    boolean isActive = status == XtmHubConnectivityStatus.ACTIVE;
+    LocalDateTime lastCheck = parseLastConnectivityCheck(settings);
+
+    return new ConnectivityCheckResult(isActive, lastCheck);
+  }
+
+  private LocalDateTime parseLastConnectivityCheck(PlatformSettings settings) {
+    String lastCheckStr = settings.getXtmHubLastConnectivityCheck();
+    return lastCheckStr != null ? LocalDateTime.parse(lastCheckStr) : LocalDateTime.now();
+  }
+
+  private void handleConnectivityLossNotification(
+      PlatformSettings settings, ConnectivityCheckResult checkResult) {
+
+    if (shouldSendConnectivityLossEmail(settings, checkResult)) {
+      xtmHubEmailService.sendLostConnectivityEmail();
+    }
   }
+
+  private boolean shouldSendConnectivityLossEmail(
+      PlatformSettings settings, ConnectivityCheckResult checkResult) {
+
+    return !checkResult.isActive()
+        && hasConnectivityBeenLostForTooLong(checkResult.lastCheck())
+        && isEmailNotificationEnabled(settings);
+  }
+
+  private boolean hasConnectivityBeenLostForTooLong(LocalDateTime lastCheck) {
```

Comment:
Simplify with somethin like this :
    return lastCheck.isBefore(LocalDateTime.now().minusHours(CONNECTIVITY_EMAIL_THRESHOLD_HOURS));